### PR TITLE
SLVS-710 "string.Format" call is redundant

### PR DESCRIPTION
**Pull request by tvsonar**

### CODE REVIEW

File: its/expected/Nancy/Nancy.ViewEngines.Razor-{2C6F51DF-015C-4DB6-B44C-0E5E4F25E2A9}-S3457.json [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+"fileVersion":  "1.1.0"
+},
+"issues":  [
+{
+"ruleId":  "S3457",
+"locations":  [
+{
+"analysisTarget":  [
+{
+"uri":  "Nancy\\src\\Nancy.ViewEngines.Razor\\CSharp\\NancyCSharpRazorCodeParser.cs",
```
Comment (dbolkensteyn) [inline its/expected/Nancy/Nancy.ViewEngines.Razor-{2C6F51DF-015C-4DB6-B44C-0E5E4F25E2A9}-S3457.json]: @tvsonar I guess you already reviewed these issues - could you comment in the PR on a few of them to acknolwedge that the issue is real or a likely false-positive

We'll be able to use the real issues found to make a blog post to announce the new release

Comment (dbolkensteyn) [inline its/expected/Nancy/Nancy.ViewEngines.Razor-{2C6F51DF-015C-4DB6-B44C-0E5E4F25E2A9}-S3457.json]: I've launched an internal email thread to collect the good results of the current sprint

File: src/SonarLint.CSharp/Rules/StringFormatWithNoArgument.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [SqaleConstantRemediation("2min")]
+    [SqaleSubCharacteristic(SqaleSubCharacteristic.Understandability)]
+    [Rule(DiagnosticId, RuleSeverity, Title, IsActivatedByDefault)]
+    [Tags(Tag.Confusing)]
+    public class StringFormatWithNoArgument : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "S3457";
+        internal const string Title = "Formatting calls should not be made on strings without formatting symbols";
```
Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Rules/StringFormatWithNoArgument.cs]: @tvsonar as discussed, I'm not convinced about this title, I'll update it in the RSPEC

Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Rules/StringFormatWithNoArgument.cs]: done

File: src/SonarLint.CSharp/Rules/StringFormatWithNoArgument.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+                    if (invocation.ArgumentList == null ||
+                        invocation.ArgumentList.Arguments.Count == 0)
+                    {
+                        return;
+                    }
+
+                    var lookup = new MethodParameterLookup(invocation, c.SemanticModel);
+                    if (!invocation.ArgumentList.Arguments.Any(arg =>
+                    {
```
Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Rules/StringFormatWithNoArgument.cs]: @tvsonar It's only january, but I doubt I'll see more horrible code this year :)

More seriously, something must be extracted from the condition - a mult-line lambda in the condition just looks way too bad

Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Rules/StringFormatWithNoArgument.cs]: also it's not really obious why this check is required

**Comment by dbolkensteyn** [comment]
LGTM, I let you update the rule title & fix the quality flaws, no need to revalidate.

Also please comment on one of the issue in the JSON that you believe is a good catch and reply on the email thread `SonarLint 1.7 new rules highlights`

File: its/expected/Nancy/Nancy.ViewEngines.Razor-{2C6F51DF-015C-4DB6-B44C-0E5E4F25E2A9}-S3457.json (hunk 1 above)
Comment (tvsonar) [inline its/expected/Nancy/Nancy.ViewEngines.Razor-{2C6F51DF-015C-4DB6-B44C-0E5E4F25E2A9}-S3457.json]: All four are real issues. The code is always something like this: `string.Format(CultureInfo.CurrentCulture, "Cannot have more than one @model statement.")`.

File: its/expected/akka.net/Akka.Remote.TestKit-{E5957C3E-2B1E-469F-A680-7953B4DEA31B}-S3457.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{
+"version":  "0.1",
+"toolInfo":  {
+"toolName":  "Microsoft (R) Visual C# Compiler",
+"productVersion":  "1.1.0",
+"fileVersion":  "1.1.0"
+},
+"issues":  [
+{
```
Comment (tvsonar) [inline its/expected/akka.net/Akka.Remote.TestKit-{E5957C3E-2B1E-469F-A680-7953B4DEA31B}-S3457.json]: This `string.Format("tried")` is probably only used to follow the pattern used in many places throughout this file.

File: its/expected/akka.net/Akka.Cluster.Sharding.Tests-{B369F314-4D37-4AC1-B4FB-64FF671AF0BC}-S3457.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
+{
+"version":  "0.1",
+"toolInfo":  {
+"toolName":  "Microsoft (R) Visual C# Compiler",
+"productVersion":  "1.1.0",
+"fileVersion":  "1.1.0"
+},
+"issues":  [
+{
```
Comment (tvsonar) [inline its/expected/akka.net/Akka.Cluster.Sharding.Tests-{B369F314-4D37-4AC1-B4FB-64FF671AF0BC}-S3457.json]: This is a real issue. The code is `var config = ConfigurationFactory.ParseString(string.Format(""));`.

**Pull request merged**